source/components/ch4_component.cpp

```diff
@@ -104,25 +104,22 @@ void CH4Component::setData( const string& varName,
     try {
          if( varName ==  D_PREINDUSTRIAL_CH4  ) {
             H_ASSERT( data.date == Core::undefinedIndex() , "date not allowed" );
-            M0 = unitval::parse_unitval( data.value_str, data.units_str, U_PPBV_CH4 );
```

Comment:
The following leads to a 'bad lexical cast'-exception (since given as unitval, not string) in the original code:
```
double value = 1.0;
Hector::unitval uval(value, U_PPBV_CH4);
Hector::message_data data(uval);
hcore.setData(CH4_COMPONENT_NAME, D_PREINDUSTRIAL_CH4, data);
```